### PR TITLE
suggestions for API extension to accommodate erc-checker

### PR DESCRIPTION
**Pull request by Timmimim**
I extended the Job Doc, which now presents the new Step "check", as well as the Job JSON example, including an example for a Check Result JSON, formatted in a similar fashion as the already included Steps.

There is also a suggestion for a new URL parameter. By default, a GET request for a particular Job will only feature limited information about the different steps of the job. That is to avoid superfluous data traffic, since the merged diffHTML, which are included as Strings in each Checker result JSON, may be pretty large (`>5mb`). 

By using the new URL parameter `details`, a user (/developer) may decide what step they want to recieve detailed information on. I suggest the option `all` to get all the details on all the steps, or instead listing each step required.

### CODE REVIEW

File: docs/job.md [hunk 1]
```diff
@@ -145,37 +147,82 @@ The overall job state can be added to the job list response:
...
+  "updatedAt": "2017-10-23T08:45:01.237Z"
 }
 ```
 
 ### URL parameters for single job view
 
 - `:id` - id of the job to be viewed
-
+- `details` - Details of steps to be loaded. By default, only `status`, `start` and `end` of any step will be loaded. 
+Can contain the following `details`: `all, validate-bag, validate_compendium, image_prepare, image_build, image_execute, `
```
[the remark this answers is not in the document]
Comment (nuest) [reply docs/job.md]: Please clarify the default "value" can be anything, e.g. "any value other than the listed ones returns only the default, e.g. `details=no`.".

File: docs/job.md [hunk 2]
```diff
@@ -145,37 +147,82 @@ The overall job state can be added to the job list response:
...
 ```json
 200 OK
 
 {
-  "id":"nkm4L",
-  "compendium_id":"a4Dnm",
-  "creation_date": Date,
+  "id": "nkm4L",
+  "compendium_id": "a4Dnm",
+  "user": "0000-0001-9757-xxxx",
```
Comment (nuest) [inline docs/job.md]: Please use the o2r user ORCID for examples: https://orcid.org/0000-0001-6021-1617

**Comment by nuest** [comment]
@jansule Please take a look at these changes if this is a workable solution from the UI perspective.

**Comment by nuest** [comment]
@Timmimim I have another request for a change: Can you rename `errorsEncountered` to simply `errors`, because unencountered errors are not a thing at all. Please also update the checker.

**Comment by nuest** [comment]
@Timmimim Can you confirm that `diffHTML` in the API and `resultHTML` in the ERC checker documentation are referring to the same content?

If so, I would prefer to use matching names. And since the properties `images` also has no `result..`, I suggest to put the HTML into a property `display.diff`, since there might be other reasonable things to display.

```JSON
    "check": {
      "status": "success",
      "images": [
        {
          "imageIndex": 0,
          "resizeOperationCode": 0,
          "compareResults": {
            "differences": 0,
            "dimension": 1290240
          }
        }
      ],
      "display": {
        "diff":  "[merged HTML with difference highlighting for images]"
      }
      "start": "2017-10-23T08:45:01.168Z",
      "end": "2017-10-23T08:45:02.193Z",
      "errors": []
    }
```

**Comment by Timmimim** [comment]
Valid points!  

`errorsEncountered` is no more, also in the checker, PR follows shortly.

As for `resultHTML` / `diffHTML`, YES, I can confirm they are the same and I only changed the namespace, I do not remember why. Both namespaces are now changed to the requested form using `display: { diff: ... }` in both Web-Api and Checker.

**Pull request merged**